bundles/org.openhab.binding.yioremote/src/main/java/org/openhab/binding/yioremote/internal/YIOremoteDockActions.java

```diff
@@ -0,0 +1,67 @@
+/**
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.yioremote.internal;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.smarthome.core.thing.binding.ThingActions;
+import org.eclipse.smarthome.core.thing.binding.ThingActionsScope;
+import org.eclipse.smarthome.core.thing.binding.ThingHandler;
+import org.openhab.core.automation.annotation.ActionInput;
+import org.openhab.core.automation.annotation.RuleAction;
+
+/**
+ * The {@link YIOremoteDockActions} is responsible for handling the action commands
+ *
+ *
+ * @author Michael Loercher - Initial contribution
+ */
+@ThingActionsScope(name = "yioremote")
+@NonNullByDefault
+public class YIOremoteDockActions implements ThingActions {
+    private @Nullable static YIOremoteDockHandler dockHandler;
```

Comment:
As this class can be instantiated for multiple Docks, if the user adds multiple Things, this must not be static. Each Action object has its own handler.

Reply:
okay but without static it shows me an error therefor i leave it as it is

Reply:
The error is caused, since you can't access a non-static field from within a static method as `sendIRCode()` is. But if you change the code as proposed in the other comment, it's gonna work: Invoking the other non-static method by using the `actions` argument.

Reply:
Can you please simple write me the complete function correctly I don't get the point to do it right. Sorry. Thanks

Reply:
solved